packages/shared/src/components/auth/AuthDefault.tsx

```diff
@@ -83,6 +85,15 @@ const AuthDefault = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [shouldLogin]);
 
+  const getFormEmail = useCallback((e: React.FormEvent) => {
+    const form = e.currentTarget as HTMLFormElement;
+    const input = Array.from(form.elements).find(
+      (el) => el.getAttribute('name') === 'email',
+    ) as HTMLInputElement;
+
+    return input?.value?.trim();
+  }, []);
```

Comment:
I extracted this because I initially wanted to use the same pattern, but turned out I couldn't.

Reply:
We do have `registerEmail` could we not use that state to sync it?

Reply:
`registerEmail` is being set on submission the same way as `setEmail`, so we can't really use it as it will be undefined when clicking on forgot password link

Reply:
Ah yes, but I mean reuse the setter from the place where you do it now?
And simply pass that with prop drilling? (just so it's one less abstraction)

Reply:
my bad, misunderstood your point -- so you would like to pass the `setRegisterEmail` or `setEmail` as a prop to `LoginForm`

Reply:
Or basically trigger the `onForgotPassword` inside `LoginForm` to pass that internal email?
Then you only need to add `email` to the callback?

Reply:
Ah, I see, I'm already doing that here (if I understand you correctly): https://github.com/dailydotdev/apps/pull/2177/files#diff-89ed52a0b28850e8d234eccd3a777f6218a989b38f297dd8e2c46387fc8d5702R103

This callback `getFormEmail` has nothing to do with the forgot password stuff, I just extracted it out of the `onEmailSignup` handler because I was planning on reusing it, but ended up not needing it.

Reply:
Oh sorry yeah misunderstood your changes then :)

Reply:
This can also go outside of component since it does not require any state.